core/sql/exp/exp_conv.cpp

```diff
@@ -1445,6 +1445,10 @@ ex_expr::exp_return_type convUnicodeToDatetime(char *target,
    }
 
    ExRaiseSqlError(heap, diagsArea, EXE_CONVERT_DATETIME_ERROR);
+   char hexstr[256];
+   if(*diagsArea)
```

Comment:
In some code paths, both heap and **diagsArea can be NULL, but *diagsArea is not NULL, that is the reason I add a safety checking 
if(*diagsArea), but yes, *diagsArea may also be possible to be NULL (although by design that is not possible IMHO). The proposed change is a very good one. i will make the change.